app/flatpak-main.c

```diff
@@ -369,6 +369,15 @@ flatpak_option_context_parse (GOptionContext     *context,
         flatpak_disable_fancy_output ();
     }
 
+  /* sudo flatpak --user ... would operate on the root user's installation,
+   * which is almost certainly not what the user intended so just consider it
+   * an error.
+   */
+  if (opt_user && running_under_sudo ())
+    return flatpak_fail_error (error, FLATPAK_ERROR,
+                               _("Refusing to operate under sudo with --user. "
+                                 "Use a root shell to operate on the root user's installation."));
```

Comment:
Maybe also mention `, or remove sudo to operate on the user installation`? Since this is probably often an accident, maybe by someone unfamiliar with the CLI (using `sudo` as a brute-force hammer is more common than I wish), it would help direct a bit more guidance to what the solution most likely to be useful in this case is.

Reply:
Good point, I reworded it.